src/service/bill_service.rs

```diff
@@ -1714,6 +1719,79 @@ impl BillServiceApi for BillService {
         }
         Ok(())
     }
+
+    async fn check_bills_timeouts(&self, now: u64) -> Result<()> {
+        let op_codes = HashSet::from([
+            BillOpCode::RequestToPay,
+            BillOpCode::OfferToSell,
+            BillOpCode::RequestToAccept,
+        ]);
+
+        let bill_ids_to_check = self
+            .store
+            .get_bill_ids_with_op_codes_since(op_codes, 0)
+            .await?;
+
+        for bill_id in bill_ids_to_check {
```

Comment:
one thing I did in my current PR (https://github.com/BitcreditProtocol/E-Bill/pull/347) is to make this check non-fallible (in a function, logging if it failed), so one bill failing doesn't rip down the whole check.

Reply:
Yes, definitely a good idea :+1: